packages/@aws-cdk/aws-s3-notifications/lib/lambda.ts

```diff
@@ -11,19 +11,25 @@ export class LambdaDestination implements s3.IBucketNotificationDestination {
   }
 
   public bind(_scope: Construct, bucket: s3.IBucket): s3.BucketNotificationDestinationConfig {
-    const permissionId = `AllowBucketNotificationsFrom${bucket.node.uniqueId}`;
+    const permissionId = `AllowBucketNotificationsFrom${bucket.node.uniqueId}To${this.fn.permissionsNode.uniqueId}`;
```

Comment:
Added the function id since the permission won't be added under the function scope anymore, causing duplicate id's when configuring two different lambda destinations on the same bucket.